far/stddlg.cpp

```diff
@@ -222,22 +343,32 @@ int GetSearchReplaceString(
 
 	Dlg->Process();
 
-	if(const auto ExitCode = Dlg->GetExitCode(); ExitCode == dlg_button_action || ExitCode == dlg_button_all)
+	if (const auto ExitCode = Dlg->GetExitCode(); ExitCode == dlg_button_action || ExitCode == dlg_button_all)
 	{
-		SearchStr = ReplaceDlg[dlg_edit_search].strData;
-		ReplaceStr = ReplaceDlg[dlg_edit_replace].strData;
-		if (Options.CaseSensitive.has_value())
-			Options.CaseSensitive = ReplaceDlg[dlg_checkbox_case].Selected == BSTATE_CHECKED;
-		if (Options.WholeWords.has_value())
-			Options.WholeWords = ReplaceDlg[dlg_checkbox_words].Selected == BSTATE_CHECKED;
-		if (Options.Reverse.has_value())
-			Options.Reverse = ReplaceDlg[dlg_checkbox_reverse].Selected == BSTATE_CHECKED;
-		if (Options.Regexp.has_value())
-			Options.Regexp = ReplaceDlg[dlg_checkbox_regex].Selected == BSTATE_CHECKED;
-		if (Options.Fuzzy.has_value())
-			Options.Fuzzy = ReplaceDlg[dlg_checkbox_fuzzy].Selected == BSTATE_CHECKED;
-		if (Options.PreserveStyle.has_value())
-			Options.PreserveStyle = ReplaceDlg[dlg_checkbox_style].Selected == BSTATE_CHECKED;
+		if (DlgItems[dlg_edit_search_hex].Flags & DIF_HIDDEN)
+		{
+			Params.SearchStr = DlgItems[dlg_edit_search_text].strData;
+		}
+		else
+		{
+			Params.SearchStr = ExtractHexString(DlgItems[dlg_edit_search_hex].strData);
+			Params.SearchBytes = HexStringToBlob(Params.SearchStr, 0);
+		}
+		Params.ReplaceStr = DlgItems[dlg_edit_replace].strData;
+		if (Params.Hex.has_value())
+			Params.Hex = DlgItems[dlg_radio_hex].Selected == BSTATE_CHECKED;
+		if (Params.CaseSensitive.has_value())
+			Params.CaseSensitive = DlgItems[dlg_checkbox_case].Selected == BSTATE_CHECKED;
+		if (Params.WholeWords.has_value())
```

Comment:
- Minor: please add empty lines between `if`s, it's somewhat hard to read one big block.

- Something like this can probably be more readable too:
```C++
const auto set_if = [&](auto& Param, int const Id)
{
	if (Param.has_value())
		Param = DlgItems[Id].Selected == BSTATE_CHECKED;
};

set_if(Params.Hex, dlg_radio_hex);
set_if(Params.CaseSensitive, dlg_checkbox_case);
...
```